cluster/cluster.go

```diff
@@ -106,7 +151,17 @@ func Join(
 		stopc:  make(chan struct{}),
 		readyc: make(chan struct{}),
 		logger: l,
+		peers:  map[string]peer{},
+	}
+
+	if reconnectInterval != 0 {
```

Comment:
It could be done at the very end of the function once it is certain that it will return without error.